### PR TITLE
language: make only needed packages visible for cross sdk imports

### PR DESCRIPTION
**Pull request by robin-da**
This gives us more control over what packages are visible in the package
database during cross sdk imports and makes sure we're not accidentally
picking up the wrong one.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -688,8 +688,8 @@ execMigrate projectOpts inFile1_ inFile2_ mbDir =
                   (NM.names $ LF.packageModules lfPkg2)
           let eqModNamesStr = map (T.unpack . LF.moduleNameString) eqModNames
           let buildOptions =
-                  [ "'--package=" <> show ("instances-" <> pkgName1, [(m, m ++ "A") | m <- eqModNamesStr]) <> "'"
-                  , "'--package=" <> show ("instances-" <> pkgName2, [(m, m ++ "B") | m <- eqModNamesStr]) <> "'"
+                  [ "'--package=" <> show ("instances-" <> pkgName1, True, [(m, m ++ "A") | m <- eqModNamesStr]) <> "'"
```
[the remark this answers is not in the document]
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: my favourite option would be to just use the ghc `package-id` flag and pass it through as ghc option.

File: compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs [hunk 2]
```diff
@@ -285,8 +286,17 @@ createProjectPackageDb opts thisSdkVer deps0 dataDeps = do
                         , optIsGenerated = True
                         , optDflagCheck = False
                         , optMbPackageName = Just instancesUnitIdStr
-                        , optHideAllPkgs = False
-                        , optPackageImports = [(unitIdStr, []) | pkgName /= "daml-stdlib"]
+                        , optHideAllPkgs = True
+                        , optPackageImports =
+                              ("daml-prim", True, []) :
+                              (unitIdStr, True, []) :
+                              -- we need the standart library from the current sdk for the
```
[the remark this answers is not in the document]
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs]: thanks

**Comment by cocreature** [comment]
Seems to fail reproducibly on CI with `Message:  <command line>: cannot satisfy instances-a-2.0(use -v for more information)`

**Comment by robin-da** [comment]
yep, working on it.

**Pull request merged**